src/Microsoft.Health.Fhir.SqlServer/Features/Watchdogs/DefragWatchdog.cs

```diff
@@ -0,0 +1,490 @@
+            using SqlConnectionWrapper conn = await scopedSqlConnectionWrapperFactory.Value.ObtainSqlConnectionWrapperAsync(cancellationToken, enlistInTransaction: false);
+            using SqlCommandWrapper cmd = conn.CreateRetrySqlCommand();
+
+            cmd.CommandText = "SELECT Number FROM dbo.Parameters WHERE Id = @Id";
+            cmd.Parameters.AddWithValue("@Id", id);
+            var value = await cmd.ExecuteScalarAsync(cancellationToken);
+
+            if (value == null)
+            {
+                throw new InvalidOperationException($"{id} is not set correctly in the Parameters table.");
+            }
+
+            return (double)value;
+        }
+
+        private async Task<int> GetThreads(CancellationToken cancellationToken)
+        {
+            var value = await GetNumberParameterById(ThreadsId, cancellationToken);
+            return (int)value;
+        }
+
+        private async Task<int> GetHeartbeatPeriod(CancellationToken cancellationToken)
+        {
+            var value = await GetNumberParameterById(HeartbeatPeriodSecId, cancellationToken);
+            return (int)value;
+        }
+
+        private async Task<int> GetHeartbeatTimeout(CancellationToken cancellationToken)
+        {
+            var value = await GetNumberParameterById(HeartbeatTimeoutSecId, cancellationToken);
+            return (int)value;
+        }
+
+        private async Task<double> GetPeriod(CancellationToken cancellationToken)
+        {
+            var value = await GetNumberParameterById(PeriodSecId, cancellationToken);
+            return value;
+        }
+
+        private async Task<bool> IsEnabled(CancellationToken cancellationToken)
+        {
+            var value = await GetNumberParameterById(IsEnabledId, cancellationToken);
+            return value == 1;
+        }
+
+        private async Task InitParamsAsync()
+        {
+            // No CancellationToken is passed since we shouldn't cancel initialization.
+
+            _logger.LogInformation("InitParams starting...");
+
+            // Offset for other instances running init
+            await Task.Delay(RandomDelay(), CancellationToken.None);
+
+            using IScoped<SqlConnectionWrapperFactory> scopedConn = _sqlConnectionWrapperFactory.Invoke();
+            using SqlConnectionWrapper conn = await scopedConn.Value.ObtainSqlConnectionWrapperAsync(CancellationToken.None, false);
+            using SqlCommandWrapper cmd = conn.CreateRetrySqlCommand();
+
+            cmd.CommandText = @"
+INSERT INTO dbo.Parameters (Id,Number) SELECT @IsEnabledId, 0
+INSERT INTO dbo.Parameters (Id,Number) SELECT @ThreadsId, 4
+INSERT INTO dbo.Parameters (Id,Number) SELECT @PeriodSecId, 24*3600
+INSERT INTO dbo.Parameters (Id,Number) SELECT @HeartbeatPeriodSecId, 60
+INSERT INTO dbo.Parameters (Id,Number) SELECT @HeartbeatTimeoutSecId, 600
+INSERT INTO dbo.Parameters (Id,Char) SELECT name, 'LogEvent' FROM sys.objects WHERE type = 'p' AND name LIKE '%defrag%'
```

Comment:
Is this needed with the code in [SqlServerFhirModel.cs](https://github.com/microsoft/fhir-server/pull/2912/files#diff-9fd99f52f9c13ba0c89f3de434555c6cd1541a6a787009d713a9c820dd1a6968)?

Reply:
Maybe server model is wrong place to do any of it. Where is a good place from which to put TEST only settings? This should be before storage ready is set.